File: CESMAC 4° periodo/POO/AV1/ESTUDOS/CURSO POO BOSON TREINAMENTOS/Como criar e usar Métodos em Classes no Python/main.py
class Gato:
    def __init__(self,nome,cor,idade):
        self.nome = nome
        self.cor = cor 
        self.idade = idade

    def peso_gato(self, peso):
        self.peso = peso
        if (self.peso > 5.0):
            print("Esta gordo")
        elif (self.peso > 3.5):
            print("Ideal")
        else:
            print("O animal esta abaixo do peso")

    def _dieta_especial_gato(self):
        self.msg = "tudo ok"
        if(self.peso < 3.5):
            self.msg = "Aumente a ração"
        if(self.peso >= 5.0):
            self.msg = "Reduza a ração" 
        return self.msg
    
    def daods_gato(self):
        print(self._dieta_especial_gato())
        return f"Nome: {self.nome}, Cor: {self.cor}, Idade: {self.idade}, Peso: {self.peso}"
        
    



if __name__ == "__main__":
    gato = Gato("Mingau", "Cinza", 2)
    print(f"Nome do gato: {gato.nome}")
    print(f"Peso do gato: {gato.peso_gato(4.0)}")
    print(gato.daods_gato())

File: CESMAC 4° periodo/POO/AV1/ESTUDOS/CURSO POO BOSON TREINAMENTOS/Como criar e usar Métodos em Classes no Python/main.java
/******************************************************************************

                            Online Java Compiler.
                Code, Compile, Run and Debug java program online.
Write your code in this editor and press "Run" button to execute it.

*******************************************************************************/

public class Main{
    public static class Gato{

        public String nome;
        public String cor;
        public int idade;

        public Gato(Stirng nome,String cor, int idade){
            this.nome = nome;
            this.cor = cor;
            this.idade = idade;
        }

        public int peso_gato(int peso){
            this.peso = peso;
            if (this.peso > 5.0){
                System.out.println("Esta gordo");
            } else if (this.peso > 3.5){
                System.out.println("Ideal");
            } else {
                System.out.println("O animal esta abaixo do peso");
            }       
            return this.peso;
        }

        

            }
        }

    }
    
	public static void main(String[] args) {
		Gato g = new Gato();
        System.out.println("Nome do gato: " + g.nome);
        System.out.println("Peso do gato: " + g.peso_gato(4.0)); 
	}
}